src/hdmf/data_utils.py

```diff
@@ -372,6 +365,11 @@ def __next__(self):
                 self.progress_bar.write("\n")  # Allows text to be written to new lines after completion
             raise StopIteration
 
+    def __reduce__(self) -> Tuple[Callable, Iterable]:
+        instance_constructor = self._from_dict
+        initialization_args = (self._to_dict(),)
+        return (instance_constructor, initialization_args)
```

Comment:
I know this might seem silly, but could you add a test where you create a dummy GenericDCI class that defines `_from_dict` and `_to_dict` and test that this `__reduce__` method does what it should?

Reply:
Done